ls1intum/Artemis
# [Exam] Exam Scores Fix

### Checklist

### Motivation and Context



There was a bug in the calculation of the exam score. The error was comparing two ids (Long)  using `==`. This caused an error when the id was above 127. Further explanation: https://stackoverflow.com/questions/20541636/comparing-boxed-long-values-127-and-128

### Description

- ID comparison is now performed using `.equals`
- Fixed 0 not displaying correctly in exam score page and csv export (0 is falsy in js, this caused this bug)

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Create an Exam with every exercise type included (**also activate manual review of programming exercises**)
4. Conduct the Exam
5. Correct exam (**also manually correct programming exercises**)
6. Open the Scores page and see if everything is displayed correctly.
7. Try the csv export and see if everything is correctly included
